### PR TITLE
Frontend/React code refactor after review

### PR DESCRIPTION
**Pull request by KacperMalachowski**
# Summary of Changes

1. Tests refactor to match new messages
2. New helper function for better test code readability (getElementWhichContain)
3. helper.js file containing test helpers functions
4. tests subdirectory containing test files and test helpers
5. Rename let to const for useState
6. Team Score in separate elements
7. Button receiving className prop, but contains default styles
8. Styles adjustment for separate elements for teams score
9. Avoid using data-testid, instead we are using screen.getByText now
10. Remove unnecessary files (README.md and robots.txt) from frontend project

### CODE REVIEW

File: frontend/smart-kickers-game/src/App.js [hunk 1]
```diff
@@ -4,39 +4,35 @@ import {Button} from "./components/button"
...
-      <center><Button >Reset game</Button></center>
+      <center>
+        <Button>Reset game</Button>
+      </center>
     </>
-  )
+  );
 }
 
 export default App
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/App.js:38]: EOF

File: frontend/smart-kickers-game/src/tests/App.spec.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  it('should update score on score message', async () => {
+    render(<App />)
+    await ws.connected
+
+    ws.send(JSON.stringify({ blueScore: 10, whiteScore: 14 }));
+
+     expect(getElementWhichContain("Blue:")).toHaveTextContent("10");
+     expect(getElementWhichContain("White:")).toHaveTextContent("14");
+  })
+})
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/tests/App.spec.js:44]: EOF

File: frontend/smart-kickers-game/src/App.css [hunk 3]
```diff
@@ -84,6 +84,11 @@ h1 {
...
   padding-top: 3rem;
-  text-align: center;
+  display: flex;
+  justify-content: center;
   font-size: 3rem;
+}
+
+.game-result-item {
+  padding: 0 3rem 0 3rem;
 }
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/App.css:94]: EOF

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
I have small talk with @Wawrzyn321 and @Lyczeq and we all agree that the whole team has to configure prettier. So please add in this PR config for prettier in `.prettierrc` file. You can check our config from previous TeamHack (it's not the best, but works) [Kyma-showcase prettierrc](https://github.com/kyma-incubator/kyma-showcase/blob/main/frontend/.prettierrc)

File: frontend/smart-kickers-game/src/App.js (hunk 1 above)
Comment (KacperMalachowski) [reply frontend/smart-kickers-game/src/App.js]: > I have small talk with @Wawrzyn321 and @Lyczeq and we all agree that the whole team has to configure prettier. So please add in this PR config for prettier in `.prettierrc` file. You can check our config from previous TeamHack (it's not the best, but works) [Kyma-showcase prettierrc](https://github.com/kyma-incubator/kyma-showcase/blob/main/frontend/.prettierrc)

Config added with adjustment for current prettier version (semicolon become semi), and files fixed with it.

File: frontend/smart-kickers-game/src/tests/App.spec.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  it('should update score on score message', async () => {
+    render(<App />);
+    await ws.connected;
+
+    ws.send(JSON.stringify({ blueScore: 10, whiteScore: 14 }));
+
+    expect(getElementWhichContain('Blue:')).toHaveTextContent('10');
+    expect(getElementWhichContain('White:')).toHaveTextContent('14');
+  });
+});
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/tests/App.spec.js:43]: These two tests are doing the same thing but with another score, mby make one test with for example blue: 1 and white 14?

Comment (KacperMalachowski) [reply frontend/smart-kickers-game/src/tests/App.spec.js:43]: Good point, for now single test should be enough. In that case I'll remove that with 1 in both score.

File: frontend/smart-kickers-game/src/components/button.js [hunk 5]
```diff
@@ -1,29 +1,10 @@
-import React from "react";
```
Comment (Wawrzyn321) [inline frontend/smart-kickers-game/src/components/button.js:1]: React (and other frontend frameworks, to think about them) usually use capitalized component names, so I'd go here with `Button.js`. The same goes for `Button.css`.

File: frontend/smart-kickers-game/src/components/Button.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import './button.css';
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/Button.js:2]: Uppercase fileName

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
Looks good, but one question, is this comment necessary for you?
```
//const socket = new WebSocket("ws://localhost:3000")
```
If not just remove it, if yes leave it and let us know.

File: frontend/smart-kickers-game/src/components/Button.js (hunk 6 above)
Comment (KacperMalachowski) [reply frontend/smart-kickers-game/src/components/Button.js]: 👍 and Fixed

File: frontend/smart-kickers-game/src/components/Button.css [hunk 7]
```diff
@@ -0,0 +1,49 @@
+:root {
+  --primary: rgba(212, 238, 194, 0.822);
+}
+
+a {
+  text-decoration: none;
+}
```
Comment (Wawrzyn321) [inline frontend/smart-kickers-game/src/components/Button.css:7]: This is a global style that will apply to all `a` elements the application. Either replace it with something more specific (like class selector), if it should be applied only for your `<Button>`s, or move it to a global css file at the application root.

File: frontend/smart-kickers-game/src/App.js [hunk 8]
```diff
@@ -1,42 +1,37 @@
...
-        Blue: {blueScore}
-          {"   "}
-        White: {whiteScore}
+      <div className="game-result-container">
+        <p className="game-result-item">Blue: {blueScore}</p>
+        <p className="game-result-item">White: {whiteScore}</p>
       </div>
-      <center><Button >Reset game</Button></center>
+      <center>
+        <Button>Reset game</Button>
```
Comment (marekkawalski) [inline frontend/smart-kickers-game/src/App.js:31]:
```suggestion
        <Button onClick={() => socket.send(JSON.stringify(true))}>
          Reset game
        </Button>
```

**Comment by Lyczeq** [comment]
I saw that you use React icons in the `src/public` directory, if you don't need them(I assume you don't), please remove them.

File: frontend/smart-kickers-game/src/App.js (hunk 8 above)
Comment (KacperMalachowski) [reply frontend/smart-kickers-game/src/App.js:31]: After a short discussion with @marekkawalski, we decided to discard this change due to near communication protocol changes (moving from WebSocket to rest API for no time-critical communication)

**Review by mrCherry97 — APPROVED** [review]

**Review by Wawrzyn321 — APPROVED** [review]

**Review by marekkawalski — APPROVED** [review]

**Pull request merged**